sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/RetryOptions.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    /**
+     * Creates an instance of {@link RetryOptions} with values for {@code fixedDelay}. Use this constructor for
+     * fixed retry delay strategy.
+     *
+     * @param maxRetries The maximum number of retry attempts to be made.
+     * @param fixedDelay The fixed delay duration between retry attempts.
+     */
+    public RetryOptions(int maxRetries, Duration fixedDelay) {
+        Objects.requireNonNull(fixedDelay, "'fixedDelay' cannot be null.");
+        if (maxRetries < 0) {
+            LOGGER.log(ClientLogger.LogLevel.VERBOSE, () -> "Max retries cannot be less than 0. " +
+                "Using 3 retries as the maximum.");
+            maxRetries = 3;
+        }
+        this.maxRetries = maxRetries;
+        this.fixedDelay = fixedDelay;
+        this.baseDelay = null;
+        this.maxDelay = null;
+    }
+
+    /**
+     * Get the maximum number of retry attempts to be made.
+     * @return the max retry attempts.
+     */
+    public int getMaxRetries() {
+        return maxRetries;
+    }
+
+    /**
+     * Get the base delay duration for retry.
+     * @return the base delay duration.
+     */
+    public Duration getBaseDelay() {
+        return baseDelay;
+    }
+
+    /**
+     * Get the max delay duration for retry.
+     * @return the max delay duration.
+     */
+    public Duration getMaxDelay() {
+        return maxDelay;
+    }
+
+    /**
+     * Get the fixed delay duration between retry attempts.
+     * @return the fixed delay duration.
+     */
+    public Duration getFixedDelay() {
+        return fixedDelay;
+    }
+
+    /**
+     * Gets the predicate that determines if a retry should be attempted.
+     * <p>
+     * If null, the default behavior is to retry HTTP responses with status codes 408, 429, and any 500 status code that
+     * isn't 501 or 505. And to retry any {@link Exception}.
+     *
+     * @return The predicate that determines if a retry should be attempted.
+     */
+    public Predicate<RequestRetryCondition> getShouldRetryCondition() {
+        return shouldRetryCondition;
+    }
+
+    /**
+     * Sets the predicate that determines if a retry should be attempted.
+     * <p>
+     * If null, the default behavior is to retry HTTP responses with status codes 408, 429, and any 500 status code that
+     * isn't 501 or 505. And to retry any {@link Exception}.
+     *
+     * @param shouldRetryCondition The predicate that determines if a retry should be attempted for the given
+     * {@link HttpResponse}.
+     * @return The updated {@link RetryOptions} object.
+     */
+    public RetryOptions setShouldRetryCondition(Predicate<RequestRetryCondition> shouldRetryCondition) {
+        this.shouldRetryCondition = shouldRetryCondition;
+        return this;
+    }
+
+    /**
+     * Gets the headers that will be added to a retry request.
+     * @return The headers that will be added to a retry request.
+     */
+    public Function<HeaderName, Duration> getRetryHeaders() {
```

Comment:
Two small things,

I think it should be `Function<Headers, Duration>`, give the caller all the `Headers` and let them make the determination instead of a specific `HeaderName`.

Also, need a small rename, maybe `backoffFromHeadersFunction` or something along those line. Ex, `getBackoffFromHeadersFunction`. And right now, the documentation makes this appear to be headers we're adding into the retry, not headers we will calculate the retry from.

Reply:
Sounds good, but should we rename it to maybe delayFromHeaders, as we havent used the backoff term in public apis and having "function" for a function return type is redundant?